[DOCS] Enhancements to Upgrade Elastic Security topic (#1580)
elastic/security-docs: docs/upgrade/upgrade-security.asciidoc

[chapter]
[[upgrade-intro]]

= Upgrade {elastic-sec}

We highly recommend that all {elastic-sec} users keep their deployments up to date with the latest available {stack} version to access new features, security updates, and performance enhancements.

For upgrading to the latest {stack} version, we recommend using the {kibana-ref}/upgrade-assistant.html[Kibana Upgrade Assistant].

NOTE: For detailed information about the 8.0.0 release, check out the <<release-notes-8.0.0, Release notes>>.

[discrete]
[[pre-upgrade-security]]
== Pre-upgrade requirements

> *Track rules that are automatically disabled when upgrading*

Reviewer: This section should match the section in this [other page](https://security-docs_1558.docs-preview.app.elstc.co/guide/en/security/master/release-notes-header-8.0.0.html) (@nastasha-solomon's PR is [here](https://github.com/elastic/security-docs/pull/1558) for reference):

<img width="800" alt="image" src="https://user-images.githubusercontent.com/616158/154387317-1a4cc00e-fd47-43a4-8b50-058d94710e3c.png">

Upon upgrading to 8.0.0, rules are automatically disabled. Once the upgrade completes, you should re-enable them again to eliminate gaps in rule coverage. _However, we highly recommend that you track your active rules before upgrading so you can easily find them and re-enable them after upgrading to 8.0._

Before upgrading, use the <<rules-api-find, Find rules API>> to capture a list of enabled rules.

IMPORTANT: You'll need to use curl or another HTTP tool to run {elastic-sec} API commands.

Here is a curl command that you can run to find enabled rules:

[source,console]
--------------------------------------------------
GET /api/detection_engine/rules/_find?per_page=10000&filter=alert.attributes.enabled:true
--------------------------------------------------

> NOTE: In the upcoming 8.0.1 release, {elastic-sec} will provide a more streamlined approach to find rules that were disabled during the upgrade process.

Author: I will remove this note.

NOTE: You do not need to shut down your {agent}s or endpoints to upgrade the {stack}.

*Requirements to display "Data views" in the {es-sec-app}*

To make the *Data view* option appear in an environment with legacy alerts, a user with elevated role privileges must visit the {es-sec-app}, open a page that displays alert data (such as the Overview page), then refresh the page. The user's role privileges must allow them to enable the detections feature in a Kibana space. Refer to <<enable-detections-ui, Enable and access detections>> for more information.

NOTE: If new alerts are generated in an upgraded environment without legacy alerts, refreshing any page with alert data in {elastic-sec} will make the *Data view* option appear in the {es-sec-ui}.

*New alert schema*

> The system index for detection alerts has been renamed from `.siem-signals-<space-id>*` to `.alerts-security.alerts-<space-id>*` and is now a hidden index. Therefore, the schema used for alert documents in {elastic-sec} has changed. Users that access documents in the `.siem-signals` indices using the {elastic-sec} API must modify their API queries and scripts to operate properly on the new 8.0 alert documents. Review the new <<alert-schema, Alert schema>>.

Reviewer 2: ```suggestion
The system index for detection alerts has been renamed from `.siem-signals-<space-id>*` to `.alerts-security.alerts-<space-id>*` and is now a hidden index. Therefore, the schema used for alert documents in {elastic-sec} has changed. Users that access documents in the `.siem-signals` indices using the {elastic-sec} API must modify their API queries and scripts to operate properly on the new 8.0.0 alert documents. Review the new <<alert-schema, Alert schema>>.
```

> The system index for detection alerts has been renamed from `.siem-signals-<space-id>*` to `.alerts-security.alerts-<space-id>*` and is now a hidden index. Therefore, the schema used for alert documents in {elastic-sec} has changed. Users that access documents in the `.siem-signals` indices using the {elastic-sec} API must modify their API queries and scripts to operate properly on the new 8.0 alert documents. Review the new <<alert-schema, Alert schema>>.

Reviewer 3: ```suggestion
The system index for detection alerts has been renamed from `.siem-signals-<space-id>*` to `.alerts-security.alerts-<space-id>*` and is now a hidden index. Therefore, the schema used for alert documents in {elastic-sec} has changed. Users that access documents in the `.siem-signals` indices using the {elastic-sec} API must modify their API queries and scripts to operate properly on the new 8.0 alert documents. You can visit <<query-alert-indices, How to query alert indices>> and review the new <<alert-schema, Alert schema>>.
```

*New privileges required to view alerts and preview rules*

> * To view alerts, users need `manage`, `write`, `read`, and `view_index_metadata` privileges to two new indices, `.alerts-security.alerts` and `.internal.alerts-security.alerts`. Existing users who are upgrading to 8.0 can retain their privileges to the `.siem-signals` index.

Reviewer 2: ```suggestion
* To view alerts, users need `manage`, `write`, `read`, and `view_index_metadata` privileges to two new indices, `.alerts-security.alerts` and `.internal.alerts-security.alerts`. Existing users who are upgrading to 8.0.0 can retain their privileges to the `.siem-signals` index.
```

* To <<preview-rules, preview rules>>, users need `read` access to the new `.preview.alerts-security.alerts` index. Refer to <<detections-permissions-section>> for more information.

> *New Full Disk Access approval for {elastic-endpoint}*

Reviewer 2: ```suggestion
*New FDA approval for {elastic-endpoint}*
```

> New Full Disk Access approval is needed when deploying {elastic-endpoint} on macOS. Refer to <<deploy-elastic-endpoint, Install {elastic-endpoint} manually>> for more information.

Reviewer 2: ```suggestion
New FDA approval is needed when deploying {elastic-endpoint} on macOS. Refer to <<deploy-elastic-endpoint, Install {elastic-endpoint} manually>> for more information.
```

*Updates to indictor match rules*

> * Changes to the indicator match rule's <<rule-ui-advanced-params, default threat indicator path>> might require you to update existing rules, or create new ones, after upgrading to {stack} version 8.0.0. Be mindful of the following:

Reviewer 2: ```suggestion
* Changes to the indicator match rule's <<rule-ui-advanced-params, default threat indicator path>> might require you to update existing rules, or create new ones, after upgrading to 8.0.0. Be mindful of the following:
```

Reviewer 2: I feel like we've repeated that these procedures are all in the effort of upgrading to 8.0, so there's no reason specify that they'd be upgrading the stack at this point in the topic.

Before upgrading, use the <<rules-api-find, Find rules API>> to capture a list of enabled rules.

IMPORTANT: Use curl or another HTTP tool to run {elastic-sec} API commands.

Here is a curl command that you can run to find enabled rules:

[source,console]
--------------------------------------------------
GET /api/detection_engine/rules/_find?per_page=10000&filter=alert.attributes.enabled:true
--------------------------------------------------

To re-enable your rules from the Rules page:

. Go to the All rules table (*Detect -> Rules*).
. Select the rules that you want to enable.
. Click *Bulk actions -> Activate Selected* to re-enable the rules.

Alternatively, you can use the <<bulk-actions-rules-api, Bulk rule actions>> API to re-enable rules.

*New Full Disk Access approval needed for {elastic-endpoint}*

When you manually install {elastic-endpoint}, you must approve a system extension, kernel extension, and enable Full Disk Access (FDA). There is a new FDA requirement in 8.0.0. Refer to <<deploy-elastic-endpoint>> to review the required permissions.

> *Upgrade agents*

Reviewer 2: Should this be "Elastic Agents" instead of just "agents"?

> Upgrade your {stack} and {agent}s to 7.17.0 first (refer to {fleet-guide}/upgrade-elastic-agent.html[Upgrade Fleet-managed Elastic Agents]). After this is complete, you can then {stack-ref}/upgrading-elastic-stack.html[upgrade the {stack}] to 8.0. Initially, {agent}s will be version 7.17.0; this is fine because {elastic-sec} supports the last minor in 7.x (7.17.0) and any subsequent {elastic-endpoint} versions in 8.x. After the {stack} upgrade, you can decide whether to upgrade {agent}s to 8.0.0, which is recommended to ensure you get the latest features and improvements.

Reviewer 4: ```suggestion
Upgrade your {stack} and {agent}s to 7.17.0 first (refer to {fleet-guide}/upgrade-elastic-agent.html[Upgrade Fleet-managed Elastic Agents]). After this is complete, you can then {stack-ref}/upgrading-elastic-stack.html[upgrade the {stack}] to 8.0.0. Initially, {agent}s will be version 7.17.0; this is fine because {elastic-sec} supports the last minor release in 7.x (7.17.0) and any subsequent {elastic-endpoint} versions in 8.x. After the {stack} upgrade, you can decide whether to upgrade {agent}s to 8.0.0, which is recommended to ensure you get the latest features and improvements.
```

*New alert schema*

The system index for detection alerts has been renamed from `.siem-signals-<space-id>` to `.alerts-security.alerts-<space-id>` and is now a hidden index. Therefore, the schema used for alert documents in {elastic-sec} has changed. Users that access documents in the `.siem-signals` indices using the {elastic-sec} API must modify their API queries and scripts to operate properly on the new 8.0.0 alert documents. Refer to <<query-alert-indices, how to query alert indices>> and review the new <<alert-schema, Alert schema>>.

*New privileges required to view alerts and preview rules*

* To view alerts, users need `manage`, `write`, `read`, and `view_index_metadata` privileges to two new indices, `.alerts-security.alerts` and `.internal.alerts-security.alerts`. Existing users who are upgrading to 8.0.0 can retain their privileges to the `.siem-signals` index.

* To <<preview-rules, preview rules>>, users need `read` access to the new `.preview.alerts-security.alerts` index. Refer to <<detections-permissions-section>> for more information.

*Full Disk Access (FDA) approval for {elastic-endpoint}*

> New FDA approval is needed when deploying {elastic-endpoint} on macOS. Refer to <<deploy-elastic-endpoint, Install {elastic-endpoint} manually>> for more information.

Reviewer 4: This section covers the same information in the previous section "New Full Disk Access approval needed for {elastic-endpoint}" (starting on line 40 above). Remove one or the other? This one seems to have the least info, so maybe remove it and keep the one above.
